src/riscv.c

```diff
@@ -20,19 +20,14 @@ static void block_map_init(block_map_t *map, const uint8_t bits)
     map->block_capacity = 1 << bits;
     map->size = 0;
     map->map = calloc(map->block_capacity, sizeof(struct block *));
-
-    map->block_mp = mpool_create(sizeof(block_t) << BLOCK_MAP_CAPACITY_BITS,
-                                 sizeof(block_t));
-    map->block_ir_mp = mpool_create(
-        sizeof(rv_insn_t) << BLOCK_IR_MAP_CAPACITY_BITS, sizeof(rv_insn_t));
 }
 
 /* clear all block in the block map */
-void block_map_clear(block_map_t *map)
+void block_map_clear(riscv_t *rv)
 {
-    assert(map);
-    for (uint32_t i = 0; i < map->block_capacity; i++) {
-        block_t *block = map->map[i];
+    block_map_t map = rv->block_map;
```

Comment:
I think this should be a direct reference(`block_map_t *map = &rv->block_map`) instead of having another instance on the stack?